chirp-impl/src/main/java/sample/chirper/chirp/impl/JpaSessionImpl.java

```diff
@@ -0,0 +1,68 @@
+package sample.chirper.chirp.impl;
+
+import akka.Done;
+import com.lightbend.lagom.internal.persistence.jdbc.SlickProvider;
+import play.inject.ApplicationLifecycle;
+import scala.compat.java8.FutureConverters;
+import scala.compat.java8.JFunction0;
+import slick.jdbc.JdbcBackend.DatabaseDef;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.EntityTransaction;
+import javax.persistence.Persistence;
+import java.util.concurrent.CompletionStage;
+import java.util.function.Function;
+
+@Singleton
+public class JpaSessionImpl implements JpaSession {
+    private final EntityManagerFactory factory;
+    private final DatabaseDef slick;
+
+    @Inject
+    JpaSessionImpl(SlickProvider slick, ApplicationLifecycle lifecycle) {
+        this.factory = Persistence.createEntityManagerFactory("default");
```

Comment:
It depends on what's being done - if you have something like Hibernate auto create tables configured, then it will fail.

It shouldn't be too hard to store the EM in a `CompletableFuture` that gets redeemed when the entity manager is successfully created, and if successfully creating it fails, retry (recursively) in X seconds.